File: DSA-loan-management/dsa-mgmt-be/app/core/enums.py
from enum import Enum


class UserRole(str, Enum):
    ADMIN = "admin"
    AGENT = "agent"
    CUSTOMER = "customer"


class LoanApplicationStatus(str, Enum):
    LEAD_CREATED = "Lead Created"
    IN_REVIEW = "In Review"
    DOCS_PENDING = "Documents Pending"
    APPROVED = "approved"
    REJECTED = "rejected"


class ContactEnquiryStatus(str, Enum):
    NEW = "New"
    IN_PROGRESS = "In Progress"
    RESOLVED = "Resolved"
    CLOSED = "Closed"


class EligibilityStatus(str, Enum):
    ELIGIBLE = "ELIGIBLE"
    PARTIALLY_ELIGIBLE = "PARTIALLY_ELIGIBLE"
    NOT_ELIGIBLE = "NOT_ELIGIBLE"
    INCOMPLETE_DETAILS = "INCOMPLETE_DETAILS"
    ERROR = "ERROR"


class LoanProductType(str, Enum):
    HOME_LOAN = "Home Loan"
    CAR_LOAN = "Car Loan"
    PERSONAL_LOAN = "Personal Loan"
    BUSINESS_LOAN = "Business Loan"
    LOAN_AGAINST_PROPERTY = "Loan Against Property"
    OTHER = "Other"


class EmploymentType(str, Enum):
    SALARIED = "Salaried"
    SELF_EMPLOYED_PROFESSIONAL = "Self Employed Professional"
    SELF_EMPLOYED_BUSINESS = "Self Employed Business"
    OTHER = "Other"


class Gender(str, Enum):
    MALE = "Male"
    FEMALE = "Female"
    OTHER = "Other"


class PropertyUsageType(str, Enum):
    RESIDENTIAL = "Residential"
    COMMERCIAL = "Commercial"


class PropertyRequirement(str, Enum):
    READY_TO_MOVE = "Ready to Move"
    UNDER_CONSTRUCTION = "Under Construction"


class PropertyType(str, Enum):
    APARTMENT = "Apartment"
    INDEPENDENT_HOUSE = "Independent House"
    PLOT = "Plot"


class PropertyStatus(str, Enum):
    FREEHOLD = "Freehold"
    LEASEHOLD = "Leasehold"


class CarCondition(str, Enum):
    NEW = "New"
    USED = "Used"

File: DSA-loan-management/dsa-mgmt-fe/src/types/enums.ts
export enum UserRole {
  ADMIN = 'admin',
  AGENT = 'agent',
  CUSTOMER = 'customer',
}

export enum LoanApplicationStatus {
  LEAD_CREATED = 'Lead Created',
  IN_REVIEW = 'In Review',
  DOCS_PENDING = 'Documents Pending',
  APPROVED = 'approved',
  REJECTED = 'rejected',
}

export enum ContactEnquiryStatus {
  NEW = 'New',
  IN_PROGRESS = 'In Progress',
  RESOLVED = 'Resolved',
  CLOSED = 'Closed',
}

export enum EligibilityStatus {
  ELIGIBLE = 'ELIGIBLE',
  PARTIALLY_ELIGIBLE = 'PARTIALLY_ELIGIBLE',
  NOT_ELIGIBLE = 'NOT_ELIGIBLE',
  INCOMPLETE_DETAILS = 'INCOMPLETE_DETAILS',
  ERROR = 'ERROR',
}

export enum LoanProductType {
  HOME_LOAN = 'Home Loan',
  CAR_LOAN = 'Car Loan',
  PERSONAL_LOAN = 'Personal Loan',
  BUSINESS_LOAN = 'Business Loan',
  LOAN_AGAINST_PROPERTY = 'Loan Against Property',
  OTHER = 'Other',
}

export enum EmploymentType {
  SALARIED = 'Salaried',
  SELF_EMPLOYED_PROFESSIONAL = 'Self Employed Professional',
  SELF_EMPLOYED_BUSINESS = 'Self Employed Business',
  OTHER = 'Other',
}

export enum Gender {
  MALE = 'Male',
  FEMALE = 'Female',
  OTHER = 'Other',
}

export enum PropertyUsageType {
  RESIDENTIAL = 'Residential',
  COMMERCIAL = 'Commercial',
}

export enum PropertyRequirement {
  READY_TO_MOVE = 'Ready to Move',
  UNDER_CONSTRUCTION = 'Under Construction',
}

export enum PropertyType {
  APARTMENT = 'Apartment',
  INDEPENDENT_HOUSE = 'Independent House',
  PLOT = 'Plot',
}

export enum PropertyStatus {
  FREEHOLD = 'Freehold',
  LEASEHOLD = 'Leasehold',
}

export enum CarCondition {
  NEW = 'New',
  USED = 'Used',
}
